presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -593,6 +594,45 @@ public static Boolean stWithin(@SqlType(GEOMETRY_TYPE_NAME) Slice left, @SqlType
         return leftGeometry.within(rightGeometry);
     }
 
+    @ScalarFunction
+    @Description("Calculates the great-circle distance between two points on the Earth's surface")
+    @SqlType(StandardTypes.DOUBLE)
+    public static double greatCircleDistance(@SqlType(StandardTypes.DOUBLE) double latitude1, @SqlType(StandardTypes.DOUBLE) double longitude1, @SqlType(StandardTypes.DOUBLE) double latitude2, @SqlType(StandardTypes.DOUBLE) double longitude2)
+    {
+        checkLatitude(latitude1);
+        checkLongitude(longitude1);
+        checkLatitude(latitude2);
+        checkLongitude(longitude2);
+
+        latitude1 = Math.toRadians(latitude1);
+        longitude1 = Math.toRadians(longitude1);
+        latitude2 = Math.toRadians(latitude2);
+        longitude2 = Math.toRadians(longitude2);
+
+        double deltaLongitude = longitude1 - longitude2;
+
+        double t1 = Math.cos(latitude2) * Math.sin(deltaLongitude);
```

Comment:
Please use descriptive variable names instead of `t1`, `t2`, ...

Reply:
@nezihyigitbasi I'm not happy about t1, t2, ..., but I can't think of descriptive names for these. I'm implementing a formula from https://en.wikipedia.org/wiki/Great-circle_distance  and using these variables only to avoid repeated computation.